pkg/storegateway/series_refs.go

```diff
@@ -904,10 +905,18 @@ type seriesHasher interface {
 	Hash(seriesID storage.SeriesRef, lset labels.Labels, stats *queryStats) uint64
 }
 
+type seriesHashCache interface {
+	Fetch(seriesID storage.SeriesRef) (uint64, bool)
+}
+
 type cachedSeriesHasher struct {
 	cache *hashcache.BlockSeriesHashCache
 }
 
+func (b cachedSeriesHasher) Fetch(seriesID storage.SeriesRef) (uint64, bool) {
+	return b.cache.Fetch(seriesID)
```

Comment:
we are. here https://github.com/grafana/mimir/blob/fd30b11fc8908322186e014388e5da941a971d43/pkg/storegateway/bucket_index_postings.go#L168

are we talking about the same place 🤔

Reply:
and now i've changed it in https://github.com/grafana/mimir/pull/3728/commits/7453753ee6247d78062cf97f51b1906ee7920e9c so that the stats are passed as a pointer to `filterPostingsByCachedShardHash` and it passes it down to `cachedSeriesHasher`, which in turn updates the stats